source/simulator/helper_functions.cc

```diff
@@ -1120,7 +1120,10 @@ namespace aspect
           for (unsigned int q=0; q<output.size(); ++q)
             {
               Tensor<1,dim> g = gravity_->gravity_vector(in.position[q]);
-              output[q] = std::fabs(std::min(0.0, g*velocity_values[q]/g.norm()))*year_in_seconds;
+              if (g.norm() < 2.0*std::numeric_limits<double>::min())
+                output[q] = 0.0;
+              else
+                output[q] = std::fabs(std::min(0.0, g*velocity_values[q]/g.norm()))*year_in_seconds;
```

Comment:
Ah, I see. It's unrelated to the rest of the patch then. How about something like
```
  const Tensor<1,dim> g = ...;
  const Tensor<1,dim> vertical = (g.norm() > 0 ? g/g.norm() : Tensor<1,dim>());
  output[q] = std::fabs(std::min(0.0, velocity_values[q]*vertical))*year_in_seconds;
```
? 

This kind of thing is best handled through a separate pull request, rather than mixing it into a bigger issue. If you'd like to, I can create such a separate request, or I can leave it to you.